### DISCUSSION TITLE
[wiki: add following page] "Scoop on Windows 7/8"

### DISCUSSION THREAD

**Discussion by goyalyashpal** [ScoopInstaller/Scoop#5885] [Show and tell]
so, i was working on a client's system some days ago. per requested me for a better package manager on per's system running x64 windows 7.

surprisingly, i was able to do that and scoop seems to work fine there too. i had made some notes on the downlaod links i used while doing that.

so, i thought that these steps might be helpful to others, the links are confirmed, but the other specifics are to the best of my memory.

-------
# Windows OS with older PowerShells

* On older versions of windows, you will need to to install some additional things to be able to update powershell.
* All the downloads and steps used here are officially from Microsoft.
* This procedure requires active online connection at one step only. All other steps can be downloaded and later transferred offline to the intended system too.
* The steps are indicative. Verbosity and spoonfeeding is not the goal here.
* These instructions are provided only in good hope, not as a guarantee or claim of any kind. These downloads may go down, or steps may stop working in future.

## Verifying if old/unsupported

[scoop-install]: https://github.com/ScoopInstaller/Install#prerequisites "Prerequisites for scoop install"

1. Run "PowerShell" (the plain one, not the x86 or ISE)
2. Run `$PSVersionTable.PSVersion`
3. If version is greater than or equal to [v5.1][scoop-install], then good.
Else, proceed with instructions below.

## Downloads

Download these things:

[ndp-452]: https://www.microsoft.com/en-us/download/details.aspx?id=42642 "NDP452-KB2901907-x86-x64-AllOS-ENU"
[wmf5download]: https://www.microsoft.com/en-us/download/details.aspx?id=54616 "Windows Management Framework (WMF) 5.1"

1. **Microsoft .NET Framework (DNF) v4.5**
Download the standalone/offline installer `NDP452-KB2901907-x86-x64-AllOS-ENU.exe` (~67 MB) from [ndp-452] page.

2. **Windows Management Framework (WMF) v5.1**
Go to [wmf5download] page, and download the zip with both `Win7` and `W2K8R2` in it, i.e. `Win7AndW2K8R2-KB3191566-x64.zip` (~65 MB) for Windows 7 SP1 x64. For Windows 8.1, download `Win8.1AndW2K12R2-KB3191564-x64.msu` (~19 MB)

## Steps for Installations

1. Run the .NET Framework's GUI installer.

2. Extract the WMF zip.

3. Run "PowerShell"

4. In the shell, navigate to directory where you extracted the zip.

5. Run `.\install-xxx.ps1`. It will likely download & install "Update for Windows(KB3191566)". 
This is the step which requires active online internet connection.

6. If it shows error "install error 80070422", then start the "Windows Update" Service aka `wuauserv` and retry.
This can be done from "Services" tab of task manager (`taskmgr`), 
or from `services.msc` window which can be found from "Start menu" > type "services".

TODO: i think there was nothing more to it, but requires confirmation.
If no more steps, then good. else add those.

7. Run `$PSVersionTable.PSVersion` to check powershell version. It should be 5.1.something.
You are good to install scoop now 😃.

## Further Notes

**Following commonly used apps do _NOT_ work in windows 7:**

* `extras/alacritty`, `extras/wezterm`: these both fundamentally use features from win10, so, are unfixable.
* `main/scoop-search` as of v1.4.1 (yeah, kinda unfortunate)
* `extras/bulk-crap-uninstaller` v5.7 - this should work when installed standalone, so requires further inspection for the cause.
* `main/yq` v4.42.1
* `main/typioca` v2.11.2
* `main/ttyper` 1.4.1 - it glitches out with something like: "winapi/winpty underline ...", so, might work with some configuration tweaks

So, suggested terminal if you will be using powershell is `extras/conemu` (v23.07.24 at time of writing).

The mintty (`extras/wsltty` v3.7.0.2) terminal is also not suggested as it:

* glitches out with powershell's autocompletion. 
* requires some changes in manifests to work standalone without `bash` shell

> It is not recommended to run with windows shell, as commands will be run using POSIX system/exec/spawn interfaces! 
> An MSYS-built [...] version will help as a solution
> https://github.com/mintty/mintty/issues/1244#issuecomment-1854872640 and https://github.com/mintty/mintty/issues/1239#issuecomment-1989606219

**Comment by xialvjun** [comment]
Does it support Windows8?

I like win8 more than 8.1.

**Reply by goyalyashpal** [reply]
> I like win8 more than 8.1.

i dont know about it. but u should try and share ur findings here 😃

**Reply by goyalyashpal** [reply]
* and if u can, try linux nixos. 

i moved to linux (nixos distro + cinammon DE) a month ago, and its soooooo much more pleasant to use these general purpose computers when:
* u dont have to walk on broken glass-shells while using them, or say when 
* the underlying OS is supporting u rather actively trying to sabotage u.
